src/main/java/net/glowstone/dispenser/ArmorDispenseBehavior.java

```diff
@@ -0,0 +1,54 @@
+package net.glowstone.dispenser;
+
+import java.util.ArrayList;
+import net.glowstone.block.GlowBlock;
+import net.glowstone.block.blocktype.BlockDispenser;
+import org.bukkit.Material;
+import org.bukkit.block.BlockFace;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.Location;
+import org.bukkit.World;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Entity;
+import java.util.List;
+import org.bukkit.entity.Player;
+import net.glowstone.inventory.GlowInventory;
+
+public class ArmorDispenseBehavior extends DefaultDispenseBehavior {
+    DefaultDispenseBehavior defaultBehavior = new DefaultDispenseBehavior();
+    
+    @Override
+    protected ItemStack dispenseStack(GlowBlock block, ItemStack stack) {
+        Material armor = stack.getType();
+        BlockFace facing = BlockDispenser.getFacing(block);
+        GlowBlock target = block.getRelative(facing);
+        Location location1 = target.getLocation();
+        World world = location1.getWorld();
+        
+        //Find all nearby entities and see if they are players or armor stands
+        List<LivingEntity> entities = new ArrayList<>();
+        for (Entity entity : world.getNearbyEntities(location1, 3, 3, 3)) {
+            switch (entity.getType()) {
+                case PLAYER: entities.add((LivingEntity) entity); break;
+                //case ARMOR_STAND: entities.add((LivingEntity) cEntity); break;
+            }
+        }  
+        
+        Location location;
+        boolean location1Test;
+        boolean location2Test;
+        //Loop through entities to see if any are in the location where armor would be dispensed
+        for (LivingEntity player : entities) {          
+            location = player.getLocation().clone();
+            location.setX(location.getBlockX());
+            location.setY(location.getBlockY());
+            location.setZ(location.getBlockZ());            
+            location1Test = (location.getX() == location1.getX() && location.getY() == location1.getY() && location.getZ() == location1.getZ());
+            location2Test = (location.getX() == player.getEyeLocation().getBlockX() && player.getEyeLocation().getBlockY() == location1.getY() && player.getEyeLocation().getBlockZ() == location1.getZ());
```

Comment:
Compare the target's location to the player's eye location. Right now, the code is comparing the player's location to their eye location.